evm/lightchain_shell.py

```diff
@@ -74,3 +76,81 @@ def cleanup():
 
 
 atexit.register(cleanup)
+
+# This is to start IPython or Python REPL.
+
+def ipython_shell_1(namespace=None, banner=None, debug=False):
+    """IPyhton v1.0 specific import"""
+    try:
+        from IPython.frontend.terminal.embed import InteractiveShellEmbed
+    except ImportError:
+        if debug:
+            traceback.print_exc()
+        return None
+    kwargs = dict(user_ns=namespace)
+    if banner:
+        kwargs = dict(banner1=banner)
+    return InteractiveShellEmbed.instance(**kwargs)
+
+
+def ipython_shell(namespace=None, banner=None, debug=False):
+    """IPython version > 1.0"""
```

Comment:
Did you consider combining those two functions in a single one? You could try importing from `IPython.terminal.embed` first and, if that fails, try from `IPython.frontend.terminal.embed`. That would avoid some duplication and we could also get rid of `start_ipython_shell` and do something like this in `start_shell()`

```
if use_ipython:
    ipython_shell = ipython_shell(...)
    if ipython_shell is not None:
        ipython_shell()
        return
    else:
        print("IPython not available")
...
```